[nnc] Fixed handling of 0-sized tensors in cat

navahgar opened the pull request:
Summary: The implementation of `aten::cat` op in NNC has to ignore tensors that have 0-size in any dimension.

Test Plan: `buck test mode/dev-nosan //caffe2/test/cpp/tensorexpr:tensorexpr -- --exact 'caffe2/test/cpp/tensorexpr:tensorexpr - Kernel.CatWithEmptyInputs'`

Differential Revision: D32122171

ZolotukhinM reviewed (approved):
LGTM! Thanks for a quick fix!

navahgar committed:
[nnc] Fixed handling of 0-sized tensors in cat (#67734)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/67734

The implementation of `aten::cat` op in NNC has to ignore tensors that have 0-size in any dimension.

Test Plan: `buck test mode/dev-nosan //caffe2/test/cpp/tensorexpr:tensorexpr -- --exact 'caffe2/test/cpp/tensorexpr:tensorexpr - Kernel.CatWithEmptyInputs'`

Reviewed By: ZolotukhinM

Differential Revision: D32122171

fbshipit-source-id: 461062bcd748a2eed8267a3d47e3477e4019ffdb
